six/modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -106,6 +112,63 @@ void readAndConvertSICD(six::NITFReadControl& reader,
         }
     }
 }
+
+six::Poly2D getXYtoRowColTransform(double center,
+                                   double sampleSpacing,
+                                   bool rowTransform)
+{
+    const double coeffs[2] = {-sampleSpacing * center, sampleSpacing};
+    const size_t orderX = rowTransform ? 1 : 0;
+    const size_t orderY = rowTransform ? 0 : 1;
+    return six::Poly2D(orderX, orderY, coeffs);
+}
+
+std::map<std::string, size_t> getAdditionalDesMap(six::NITFReadControl& reader)
+{
+    std::map<std::string, size_t> nameToDesIndex;
+    nitf::List des = reader.getRecord().getDataExtensions();
+    nitf::ListIterator desIter = des.begin();
+
+    for (size_t ii = 0; desIter != des.end(); ++desIter, ++ii)
+    {
+        // Skip the first DES. By convention, this is the SICD XML
```

Comment:
Is this guaranteed to be true per the SICD spec?  If not, you need to parse DESID to figure it out

Reply:
According to (the sparse amount of NITF-specific information present in) the SICD spec, the XML metadata is stored in the first DES. No other DES is called out in the spec. I'll add to the comment to call out the spec explicitly.

Reply:
Did you look at Volume 2 (that's where all the NITF-specific stuff is): http://www.gwg.nga.mil/ntb/baseline/docs/SICD/SICD_Volume_2.pdf